src/redux/books/booksSlice.js

```diff
@@ -1,23 +1,55 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, nanoid } from '@reduxjs/toolkit';
 
 const initialState = {
-  books: [],
+  books: [
+    {
+      id: 'item1',
+      title: 'The Great Gatsby',
+      author: 'John Smith',
+      category: 'Fiction',
+    },
+    {
+      id: 'item2',
+      title: 'Anna Karenina',
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
 };
 
 const booksSlice = createSlice({
   name: 'books',
   initialState,
   reducers: {
-    addBook: (state, action) => {
-      state.books.push(action.payload);
+    addBook: {
+      reducer: (state, action) => {
+        state.books.push(action.payload);
+      },
```

Comment:
- [ ] [OPTIONAL] It is considered best practice to not mutate the original state, especially in Redux reducers, immutability is very valuable, so try to use Array.concat() or the spread operator(...args) rather than push for adding the books, using spread operator or concat in Redux reducers instead of push ensures immutability, maintains pure functions and can improve performance when working with nested objects or arrays.

Reply:
@ PinkMoon25, redux users immer under the hood when used inside createSlice as such it generally takes care of the mutability. what do you think of that?

Reply:
Hi @zeff96, you are right it does not mutate the state, but I think practicing immutability wherever possible will help you write better code as it is needed in react's useState and other hooks too, just that immutability has a lot of use cases, and either you alway work with immer or practice immutability, I think using immer would be good if you have really complex nested data else no dependency is better 👍 
Also your question's answer by chatGpt 👇 
"While using Immer with createSlice can handle the mutability of push in a safe way, it's still recommended to use immutable operations like concat or the spread operator whenever possible to maintain consistency with Redux's principles of immutability."

Reply:
@PinkMoon25, Yeah noted, I do use a spread operator or concat while using reducers on react on simple state updates where I do not need to install immer, Thanks for the advice.

Reply:
@PinkMoon25, Explained beyond reasonable doubt, am refactoring the  code to use concat